superset/connectors/sqla/models.py

```diff
@@ -1007,6 +1007,8 @@ def adhoc_column_to_sqla(  # pylint: disable=too-many-locals
                     qry = sa.select([sqla_column]).limit(1).select_from(tbl)
                     sql = self.database.compile_sqla_query(qry)
                     col_desc = get_columns_description(self.database, self.schema, sql)
+                    if not col_desc:
```

Comment:
dealing with case where `custom_sql` can be empty string and col_desc is empty

Reply:
I think we should handle this on frontend and not let users save a custom sql that's an empty string

Reply:
@kgabryje I agree - however, as it will be lots of work to clean up old metadata, I think we can maybe have both?

Reply:
I think we should handle this edge case in the backend anyways. I will be nice to prevent user from saving empty custom sql in the frontend and show some friendly message. But for now if the custom_sql is any arbitrary string the API responds with 500 with error message directly from DB engine, which is ok and readable for users but just this empty string it throws `index out of bound error` to user which is not helpful.

Reply:
we can create a separate PR for UI messaging and prevention though